modules/text/src/ocr_holistic.cpp

```diff
@@ -0,0 +1,1064 @@
+//******************   ImagePreprocessor   *******************************************
+//************************************************************************************
+
+void ImagePreprocessor::preprocess(InputArray input,OutputArray output,Size sz,int outputChannels){
+    Mat inpImg=input.getMat();
+    Mat outImg;
+    this->preprocess_(inpImg,outImg,sz,outputChannels);
+    outImg.copyTo(output);
+}
+void ImagePreprocessor::set_mean(Mat mean){
+
+
+    this->set_mean_(mean);
+
+}
+
+
+
+class ResizerPreprocessor: public ImagePreprocessor{
+protected:
+    void preprocess_(const Mat& input,Mat& output,Size outputSize,int outputChannels){
+        //TODO put all the logic of channel and depth conversions in ImageProcessor class
+        CV_Assert(outputChannels==1 || outputChannels==3);
+        CV_Assert(input.channels()==1 || input.channels()==3);
+        if(input.channels()!=outputChannels)
+        {
+            Mat tmpInput;
+            if(outputChannels==1){
+                cvtColor(input,tmpInput,COLOR_BGR2GRAY);
+                if(input.depth()==CV_8U)
+                {
+                    tmpInput.convertTo(output,CV_32FC1,1/255.0);
+                }else
+                {//Assuming values are at the desired [0,1] range
+                    tmpInput.convertTo(output, CV_32FC1);
+                }
+            }else
+            {
+                cvtColor(input,tmpInput,COLOR_GRAY2BGR);
+                if(input.depth()==CV_8U)
+                {
+                    tmpInput.convertTo(output,CV_32FC3,1/255.0);
+                }else
+                {//Assuming values are at the desired [0,1] range
+                    tmpInput.convertTo(output, CV_32FC3);
+                }
+            }
+        }else
+        {
+            if(input.channels()==1)
+            {
+                if(input.depth()==CV_8U)
+                {
+                    input.convertTo(output, CV_32FC1,1/255.0);
+                }else
+                {//Assuming values are at the desired [0,1] range
+                    input.convertTo(output, CV_32FC1);
+                }
+            }else
+            {
+                if(input.depth()==CV_8U){
+                    input.convertTo(output, CV_32FC3,1/255.0);
+                }else
+                {//Assuming values are at the desired [0,1] range
+                    input.convertTo(output, CV_32FC3);
+                }
+            }
+        }
+        if(outputSize.width!=0 && outputSize.height!=0)
+        {
+            resize(output,output,outputSize);
+        }
+    }
+    //void set_mean_(Mat m){}
+public:
+    ResizerPreprocessor(){}
+    ~ResizerPreprocessor(){}
+};
+
+class StandarizerPreprocessor: public ImagePreprocessor{
+protected:
+    double sigma_;
+    //void set_mean_(Mat M){}
+
+    void preprocess_(const Mat& input,Mat& output,Size outputSize,int outputChannels){
+
+        //TODO put all the logic of channel and depth conversions in ImageProcessor class
```

Comment:
can you just call `ResizerPreprocessor`'s preprocess to avoid repeating all this code?